From 61a7daa00a824a165eeccf0bf3bbdd5ade0f3911 Mon Sep 17 00:00:00 2001
From: Jonas Witschel <diabonas@archlinux.org>
Date: Thu, 19 Dec 2019 13:24:22 +0100
Subject: [PATCH] Disable mouse tracking for CompletingEdit

This is a temporary workaround for a regression in Qt 5.14.0, see
https://github.com/TeXworks/texworks/issues/883 and
https://bugreports.qt.io/browse/QTBUG-80831.
---
 src/CompletingEdit.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/CompletingEdit.cpp b/src/CompletingEdit.cpp
index a3089dbd..e6c16213 100644
--- a/src/CompletingEdit.cpp
+++ b/src/CompletingEdit.cpp
@@ -103,6 +103,9 @@ CompletingEdit::CompletingEdit(QWidget *parent /* = nullptr */)
 	updateLineNumberAreaWidth(0);
 	updateColors();
 	TWUtils::installCustomShortcuts(this);
+
+	// Workaround for Qt 5.14, see https://bugreports.qt.io/browse/QTBUG-80831
+	setMouseTracking(false);
 }
 
 void CompletingEdit::prefixLines(const QString &prefix)
-- 
2.25.1

